### PR TITLE
1.10/joint.sdf: better default limit values

### PR DESCRIPTION
**Pull request by scpeters**
# 🦟 Bug fix

Part of #169.

## Summary

* Use `±inf` instead of `±1e16` for position limits
* Use `inf` instead of `-1` for effort, velocity limits

This will require a small change in gz-physics, as illustrated by the test failure in https://github.com/gazebosim/gz-physics/pull/401.

## Checklist
- [X] Signed all commits for DCO
- [ ] Added tests
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [X] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [X] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/JointAxis.cc [hunk 1]
```diff
@@ -142,11 +143,12 @@ Errors JointAxis::Load(ElementPtr _sdf)
   {
     sdf::ElementPtr limitElement = _sdf->GetElement("limit");
 
-    this->dataPtr->lower = limitElement->Get<double>("lower", -1e16).first;
-    this->dataPtr->upper = limitElement->Get<double>("upper", 1e16).first;
-    this->dataPtr->effort = limitElement->Get<double>("effort", -1).first;
+    const double kInf = std::numeric_limits<double>::infinity();
+    this->dataPtr->lower = limitElement->Get<double>("lower", -kInf).first;
```
Comment (chapulina) [inline src/JointAxis.cc:147]: You could pass the variable itself here as the default value, so if it ever changes again, it only needs to be changed on line 64, i.e.

```suggestion
    this->dataPtr->lower = limitElement->Get<double>("lower", this->dataPtr->lower).first;
```

**Review by chapulina — APPROVED** [review]

File: src/JointAxis.cc (hunk 1 above)
Comment (scpeters) [reply src/JointAxis.cc]: I've deduplicated the default values used in `JointAxis::Load` in https://github.com/gazebosim/sdformat/pull/1112/commits/c7feca99e73fdc684a9bc4705510274294d09ef5

**Comment by scpeters** [comment]
while deduplicating some default value constants, I noticed an unused variable corresponding to the deprecated and removed `use_parent_model_frame` element, so I removed that as well in https://github.com/gazebosim/sdformat/pull/1112/commits/95b3963d430eda20c301135dc53f3f2022d34710

File: sdf/1.10/joint.sdf [hunk 2]
```diff
@@ -74,16 +74,16 @@
...
-      <element name="lower" type="double" default="-1e16" required="1">
+      <element name="lower" type="double" default="-inf" required="1">
         <description>Specifies the lower joint limit (radians for revolute joints, meters for prismatic joints). Omit if joint is continuous.</description>
       </element>
-      <element name="upper" type="double" default="1e16" required="1">
+      <element name="upper" type="double" default="inf" required="1">
         <description>Specifies the upper joint limit (radians for revolute joints, meters for prismatic joints). Omit if joint is continuous.</description>
       </element>
-      <element name="effort" type="double" default="-1" required="0">
+      <element name="effort" type="double" default="inf" required="0">
```
Comment (azeey) [inline sdf/1.10/joint.sdf:83]: The description says "Limit is not enforced if value is negative". Something to 
keep in mind when updating gz-physics.

**Review by azeey — APPROVED** [review]

**Comment by azeey** [comment]
We'll also need to update sdformat-mjcf https://github.com/gazebosim/gz-mujoco/blob/fd5dec325052b0db93fadc88e667987d27bf378b/sdformat_mjcf/src/sdformat_mjcf/sdformat_to_mjcf/converters/joint.py#L29

File: sdf/1.10/joint.sdf (hunk 2 above)
Comment (scpeters) [reply sdf/1.10/joint.sdf:83]: this was already part of the behavior of the DOM object as of https://github.com/gazebosim/sdformat/pull/357

**Comment by scpeters** [comment]
> We'll also need to update sdformat-mjcf https://github.com/gazebosim/gz-mujoco/blob/fd5dec325052b0db93fadc88e667987d27bf378b/sdformat_mjcf/src/sdformat_mjcf/sdformat_to_mjcf/converters/joint.py#L29

opened a PR https://github.com/gazebosim/gz-mujoco/pull/106

but its tests are failing

**Pull request merged**